### PR TITLE
use sparse_set

### LINKED ISSUE #215: Performance: replace std::map with a fast hash-map in DefaultDatasetBackend
it's sooo slow 🐌

### PR DESCRIPTION
**Pull request by mcb5637**
close #215

### CODE REVIEW

File: src/rdf4cpp/rdf/storage/tuple/DefaultDatasetBackend.hpp [hunk 1]
```diff
@@ -18,7 +20,7 @@ struct DefaultDatasetBackend : public IDatasetBackend {
 
 public:
     // TODO: thread safety
-    std::set<Quad> quads_{};
+    dice::sparse_map::sparse_set<Quad, dice::hash::DiceHash<Quad>> quads_{};
```
Comment (liss-h) [inline src/rdf4cpp/rdf/storage/tuple/DefaultDatasetBackend.hpp]: Use wyhash (called `DiceHashWhyhash`, i think) (default is robin_hood)

File: src/rdf4cpp/rdf/storage/tuple/DefaultSolutionSequenceBackend.hpp [hunk 2]
```diff
@@ -2,17 +2,20 @@
 #define RDF4CPP_DEFAULTSOLUTIONSEQUENCEBACKEND_HPP
```
Comment (liss-h) [inline src/rdf4cpp/rdf/storage/tuple/DefaultSolutionSequenceBackend.hpp:2]: Missing `#ifndef`

**Review by liss-h — CHANGES_REQUESTED** [review]
Some minor things

File: src/rdf4cpp/rdf/storage/tuple/DefaultSolutionSequenceBackend.hpp (hunk 2 above)
Comment (mcb5637) [reply src/rdf4cpp/rdf/storage/tuple/DefaultSolutionSequenceBackend.hpp:2]: it is actually there, just the part that github shows does start at line 2

Comment (liss-h) [reply src/rdf4cpp/rdf/storage/tuple/DefaultSolutionSequenceBackend.hpp:2]: ups, sorry about that

**Review by liss-h — APPROVED** [review]

**Pull request merged**